Apply missing changes for 4.13.4

tunniclm opened the pull request:
package.json:
- Update dependencies and devDependencies and
- Change the version to 4.13.4
- Alphabetical ordering of some lists

History.md:
- Add details of updates to deps in 4.13.4

---

Related to: #2861
Related to: #2862 

I'm hoping this PR will be helpful to including the deps updates from the NPM release 4.13.4 into the main line. Let me know any comments so I can fix it as necessary.

jordonias committed:
add Gitter link to Readme.md

Commit:
Merge pull request #2848 from jordonias/master

add Gitter link to Readme.md

jasnell commented:
LGTM

ritch commented:
@tunniclm can you split these changes out into separate commits to be consistent with these sorts of changes in the past?

https://github.com/tunniclm/express/commits/master

tunniclm commented:
@ritch Sure :)

tunniclm committed:
deps: content-disposition@0.5.1

tunniclm committed:
deps: depd@~1.1.0

tunniclm committed:
deps: escape-html@~1.0.3

tunniclm committed:
deps: finalhandler@0.4.1

tunniclm committed:
deps: merge-descriptors@1.0.1

tunniclm committed:
deps: methods@~1.1.2

tunniclm committed:
deps: parseurl@~1.3.1

tunniclm committed:
deps: proxy-addr@~1.0.10

tunniclm committed:
deps: range-parser@~1.0.3

tunniclm committed:
deps: send@0.13.1

tunniclm committed:
deps: serve-static@~1.10.2

tunniclm committed:
build: istanbul@0.4.2

tunniclm committed:
build: body-parser@~1.14.2

tunniclm committed:
build: cookie-parser@~1.4.1

tunniclm committed:
build: express-session@~1.13.0

tunniclm committed:
build: mocha@2.3.4

tunniclm committed:
Order lists alphabetically

tunniclm committed:
Update version to 4.13.4

NOTE: Version 4.14.4 was originally released to NPM based on
the expressjs/express repository (a, now deleted, github fork
of this repository which appears to have been based on
commit 4fb3653b6ce9bd9080ceadb267f2cffc32413444).

The following commits (ancestors to this one) recreate the
changes to History.md and package.json:
f9b4654050aee6a51b41dec91e4f6e872d5d438e Order lists alphabetically
99573b701eb97c59a8c03e27e12a9ce9cd4c7566 build: mocha@2.3.4
04ba299984ee5c012d93ee0e6d1438bd0da84c22 build: express-session@~1.13.0
65be7f32993b91a827d7943c591a021e9807e080 build: cookie-parser@~1.4.1
25e11dbba60587b0c5074d14ecb9aa0832f2f882 build: body-parser@~1.14.2
65d00eb1d296f1462fb2451e6d1bcfaa0d234fdc build: istanbul@0.4.2
0ab598454a48159035ae08935016c30bdb5303ee deps: serve-static@~1.10.2
69b0ff54d5e741366df992d434141c7f31c9c6dc deps: send@0.13.1
6255bec69ad9c0a058bf604d3b4a3856cb559c69 deps: range-parser@~1.0.3
66140b403ae7b0c452a34b9d4f9443041e5ee196 deps: proxy-addr@~1.0.10
31e73fc1cd37d3a7f4bdcddca805e848b0ec559d deps: parseurl@~1.3.1
40744efb32cddeb1d6ae4ba674a3adb1ed07ab05 deps: methods@~1.1.2
4e333fbc37099d7c53356c88eaf2dabc44fdaca8 deps: merge-descriptors@1.0.1
914fb8fbbd42d209a6ef557e8d9cc2d63b1fb369 deps: finalhandler@0.4.1
1827559e4453f85ed9faea9914e6b5e5a7ab9929 deps: escape-html@~1.0.3
fcefa48bb0890acfd2f1868411bccdb51d96240a deps: depd@~1.1.0
47dd5d6aed9cf7a5c3638ef84c04b2c3a590e479 deps: content-disposition@0.5.1

There were also changes to Readme.md, but these have not been
included in this series of commits as they change a number of
links to point to the now deleted repository (see #2861 and #2862).

This series of commits should move the repository forward to
a point where we can be happy most of the changes from 4.13.4
are included which is a prerequisite to releasing the next
release.

I hope this helps with any potential future confusion regarding
the differences between the npm package for 4.13.4 and what is
in this repository (for example, the differences to Readme.md,
the missing/different commit hashes, etc).

tunniclm commented:
@ritch updated

tunniclm commented:
Note, commit dbdca77d84d97c36d1c57b808d20df7cb4289738 includes a detailed message regarding the commit series.

sam-github commented:
Perhaps the last commit should not update the version in the package.json, since its not derived from the actual 4.13.4 commit. and doesn't have the same SHA or the same exact contents as the 4.13.4 published version. If @dougwilson still has the commit he published from on one of his machines, he can push that to this repo, or a fork of this repo, and it can be tagged, and then all npmjs publishes will have a matching commit and tag, and history will be consistent.

Either way, perhaps the last commit should be changed to 'Update version to 4.13.5', and it could get immediately tagged and published. Then the latest published package, `.5`. points to a SHA in the express repo, and that last `.4` patch just becomes a historic oddity.

jordonias commented:
**History.md** should probably be updated as well

tunniclm commented:
@sam-github , @jordonias : Agreed. The 2 potential ways to go then would be: (a) get @dougwilson 's original commits, if he still has them (I prefer this, for the historical consistency and being able to use a correct tag that matches the npm release), or (b) don't have release 4.14.4 in this repo at all (no tag, no exact commit for that version) and make a new release 4.13.5 based on commits like those currently in this pr that merge the relevant updates in. (And so we'd have a History.md without 4.14.4, or maybe a note about why it is skipped in this repo).

I hope that summarises things accurately -- comments welcome!

tunniclm commented:
If we don't get a commit from @dougwilson, would this pr be okay to merge? Is there anything else that is missing, or just the version change in the package.json and History.md?

tunniclm commented:
@dougwilson : Are you able to provide the original 4.13.4 release commit?

If not, I propose we merge (with the version update) this and make a new release 4.13.5 on Thursday.

Fishrock123 commented:
> @dougwilson : Are you able to provide the original 4.13.4 release commit?

@tunniclm I wouldn't count on it.

dougwilson commented:
I uploaded them to the repository in #2862 two weeks ago.

dougwilson commented:
Commits has been pushed up from @ljharb's fork.

ljharb commented:
Thank you!

tunniclm commented:
@dougwilson I didn't realise you had pushed the commit somewhere or when. There was a mention of doing it in the other issue, but I didn't spot a comment saying it was done, or a link to where it was pushed.

Anyway, it's all good now! Thanks

dougwilson commented:
Hi @tunniclm, sorry there was no link. I think it was mentioned they were there in https://github.com/expressjs/express/issues/2862#issuecomment-177064473 and sorry again for no link, I guess there was an assumption of where a forked repo would end up being (and the pushes were in my public GitHub activity feed).
